test/parallel/test-process-chdir.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+assert.notStrictEqual(process.cwd(), __dirname);
+process.chdir(__dirname);
+assert.strictEqual(process.cwd(), __dirname);
+
+const dir = path.resolve(common.tmpDir,
+    'weird \uc3a4\uc3ab\uc3af characters \u00e1\u00e2\u00e3');
+
+// Make sure that the tmp directory is clean
+common.refreshTmpDir();
```

Comment:
@Fishrock123 refresh just removes the directory and creates it 
 
```js
exports.refreshTmpDir = function() {
  rimrafSync(exports.tmpDir);
  fs.mkdirSync(exports.tmpDir);
};
```

The name is actually decided when we load the module itself,

```js
if (process.env.TEST_THREAD_ID) {
  // Distribute ports in parallel tests
  if (!process.env.NODE_COMMON_PORT)
    exports.PORT += +process.env.TEST_THREAD_ID * 100;

  exports.tmpDirName += '.' + process.env.TEST_THREAD_ID;
}
exports.tmpDir = path.join(exports.testDir, exports.tmpDirName);
```
Since we are just resolving the path before that, we are safe here.